spec/request_handler/body_parser_spec.rb

```diff
@@ -3,17 +3,28 @@
 require 'request_handler/body_parser'
 describe RequestHandler::BodyParser do
   let(:handler) do
-    described_class.new(schema: schema, request: build_mock_request(params: {}, headers: {}, body: raw_body))
+    described_class.new(
+      schema:           schema,
+      request:          build_mock_request(params: {}, headers: {}, body: raw_body),
+      included_schemas: included_schemas
+    )
   end
   shared_examples 'flattens the body as expected' do
-    it 'returns the flattend body' do
-      expect(handler).to receive(:validate_schema).with(wanted_result)
+    it 'returns the flattened body' do
```

Comment:
I'd prefer if we either have a separate shared_example for the case with included schemas or - as it's only a single example right now - move this handling directly there